epochStart/bootstrap/storageProcess_test.go

```diff
@@ -125,3 +132,463 @@ func TestStorageEpochStartBootstrap_BootstrapMetablockNotFound(t *testing.T) {
 	assert.Equal(t, process.ErrNilMetaBlockHeader, err)
 	assert.Equal(t, uint32(0), params.Epoch)
 }
+
+func TestStorageEpochStartBootstrap_requestAndProcessFromStorage(t *testing.T) {
+	t.Parallel()
+
+	coreComp, cryptoComp := createComponentsForEpochStart()
+	args := createMockStorageEpochStartBootstrapArgs(coreComp, cryptoComp)
+	args.GenesisNodesConfig = getNodesConfigMock(1)
+
+	prevPrevEpochStartMetaHeaderHash := []byte("prevPrevEpochStartMetaHeaderHash")
+	prevEpochStartMetaHeaderHash := []byte("prevEpochStartMetaHeaderHash")
+	notarizedShardHeaderHash := []byte("notarizedShardHeaderHash")
+	epochStartMetaBlockHash := []byte("epochStartMetaBlockHash")
+	prevNotarizedShardHeaderHash := []byte("prevNotarizedShardHeaderHash")
+	notarizedShardHeader := &block.Header{
+		PrevHash: prevNotarizedShardHeaderHash,
+	}
+	prevNotarizedShardHeader := &block.Header{}
+	notarizedMetaHeaderHash := []byte("notarizedMetaHeaderHash")
+	prevMetaHeaderHash := []byte("prevMetaHeaderHash")
+	notarizedMetaHeader := &block.MetaBlock{
+		PrevHash: prevMetaHeaderHash,
+	}
+
+	t.Run("request and process for shard", func(t *testing.T) {
+		t.Parallel()
+
+		shardId := uint32(0)
+		args.DestinationShardAsObserver = shardId
```

Comment:
here you are changing the same resource as in the other parallel tests.
this is not concurrent safe and can generate different results
you can move the args (and maybe also the coreComp and cryptoComp) creation inside the t.Run contexts.

Reply:
right :+1: thanks